index.d.ts

```diff
@@ -0,0 +1,131 @@
+/* =================== USAGE ===================
+
+    import express = require("express");
+    var app = express();
+
+ =============================================== */
+
+/// <reference types="./core" />
+/// <reference types="serve-static" />
+
+import * as bodyParser from "body-parser";
+import * as core from "./core";
+// import * as qs from "qs"; // @see https://github.com/expressjs/discussions/issues/192#issuecomment-2708756867
+import * as serveStatic from "serve-static";
+
+/**
+ * Creates an Express application. The express() function is a top-level function exported by the express module.
+ */
+declare function e(): core.Express;
+
+declare namespace e {
+  /**
+   * This is a built-in middleware function in Express. It parses incoming requests with JSON payloads and is based on body-parser.
+   * @since 4.16.0
+   */
+  var json: typeof bodyParser.json;
+
+  /**
+   * This is a built-in middleware function in Express. It parses incoming requests with Buffer payloads and is based on body-parser.
+   * @since 4.17.0
+   */
+  var raw: typeof bodyParser.raw;
+
+  /**
+   * This is a built-in middleware function in Express. It parses incoming requests with text payloads and is based on body-parser.
+   * @since 4.17.0
+   */
+  var text: typeof bodyParser.text;
+
+  /**
+   * These are the exposed prototypes.
+   */
+  var application: Application;
+  var request: Request;
+  var response: Response;
+
+  /**
+   * This is a built-in middleware function in Express. It serves static files and is based on serve-static.
+   */
+  var static: serveStatic.RequestHandlerConstructor<Response>;
+
+  /**
+   * This is a built-in middleware function in Express. It parses incoming requests with urlencoded payloads and is based on body-parser.
+   * @since 4.16.0
+   */
+  var urlencoded: typeof bodyParser.urlencoded;
+
+  /**
+   * This is a built-in middleware function in Express. It parses incoming request query parameters.
+   * @todo this seems to be outdated, not implemented and should be removed
+   * @link https://github.com/DefinitelyTyped/DefinitelyTyped/pull/43217
+   * @link https://github.com/expressjs/discussions/issues/192#issuecomment-2708756867
+   */
+  // export function query(options: qs.IParseOptions | typeof qs.parse): Handler;
```

Comment:
this one needs to be clarified, because ATTW noticed that there is no such thing anymore as `express.query` in the source code

Reply:
found the proof https://github.com/expressjs/express/blob/master/History.md#500-alpha1--2014-11-06